### ISSUE TITLE
Support OpenSearch

### ISSUE THREAD

**Issue by netom** [zentity-io/zentity#99]
We've moving from Elasticsearch 7.10.2 to OpenSearch 1.2.4. The OS version is a fork of the OSS build of the ES version plus a bunch of plugins.

After skimming through the plugin migration documentation [(https://github.com/opensearch-project/opensearch-plugins/blob/main/UPGRADING.md)], I managed to build Zenity for OS.

Here is a proof of concept:

https://github.com/netom/zentity/tree/opensearch-1.2.4

I wonder if it would be possible for the project to be built for both Elasticsearch and OpenSearch.

I imagine this would require a complete re-design of the interface between the core functionality and the back end.

**Comment by austince** [comment]
I don't believe this will be supported, per https://github.com/zentity-io/zentity/issues/90#issuecomment-936522060

I think the suggestion would be to fork zentity as well.

**Comment by davemoore-** [comment]
Closing because I'm only going to support the official distribution of Elasticsearch (#90). OpenSearch has its own Java client libraries ([here](https://mvnrepository.com/artifact/org.opensearch.client/opensearch-java)), which would require building and documenting two very different implementations. It would be too much for me to support.
